### PR TITLE
[BEAM-632] Fix Flatten PCollections in Dataflow

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by tgroh** [comment]
R: @dhalperi

**Comment by tgroh** [comment]
Run Dataflow RunnableOnService

File: runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/PTransformMatchers.java [hunk 1]
```diff
@@ -178,6 +182,28 @@ public boolean matches(AppliedPTransform<?, ?, ?> application) {
     };
   }
 
+  /**
+   * A {@link PTransformMatcher} which matches a {@link Flatten.PCollections} which
+   * consumes no input {@link PCollection PCollections}.
```
Comment (dhalperi) [inline runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/PTransformMatchers.java]: javadoc wrong

**Comment by tgroh** [comment]
It should be stable. Done.

**Pull request closed without merging**